packages/xo-server/src/xo-mixins/proxies.js

```
@@ -342,13 +342,26 @@ export default class Proxy {
       xoaUpgradeTimeout
     )
 
-    await this.checkProxyHealth(proxyId)
+    const { success, error } = await this.checkProxyHealth(proxyId)
+    if (!success) {
+      throw error
+    }
 
     return proxyId
   }
 
   async checkProxyHealth(id) {
-    await this.callProxyMethod(id, 'system.getServerVersion')
+    try {
+      await this.callProxyMethod(id, 'system.getServerVersion')
+      return {
+        success: true,
+      }
+    } catch (error) {
+      return {
+        success: false,
+        error,
+      }
+    }
```

Simply let it throw and handle the wrapping in `api/proxy.js`.